lib-ee/emqx_ee_schema_registry/test/emqx_ee_schema_registry_SUITE.erl

```diff
@@ -731,3 +746,113 @@ t_import_config(_Config) ->
         {ok, #{root_key => schema_registry, changed => [Path]}},
         emqx_ee_schema_registry:import_config(RawConf1)
     ).
+
+sparkplug_example_data_base64() ->
+    <<"CPHh67HrMBIqChxjb3VudGVyX2dyb3VwMS9jb3VudGVyMV8xc2VjGPXh67HrMCACUKgDEikKHGNvdW50ZXJfZ3JvdXAxL2NvdW50ZXIxXzVzZWMY9eHrseswIAJQVBIqCh1jb3VudGVyX2dyb3VwMS9jb3VudGVyMV8xMHNlYxj14eux6zAgAlAqEigKG2NvdW50ZXJfZ3JvdXAxL2NvdW50ZXIxX3J1bhj14eux6zAgBVABEioKHWNvdW50ZXJfZ3JvdXAxL2NvdW50ZXIxX3Jlc2V0GPXh67HrMCAFUAAYWA==">>.
+
+sparkplug_example_data() ->
+    #{
+        <<"metrics">> =>
+            [
+                #{
+                    <<"datatype">> => 2,
+                    <<"int_value">> => 424,
+                    <<"name">> => <<"counter_group1/counter1_1sec">>,
+                    <<"timestamp">> => 1678094561525
+                },
+                #{
+                    <<"datatype">> => 2,
+                    <<"int_value">> => 84,
+                    <<"name">> => <<"counter_group1/counter1_5sec">>,
+                    <<"timestamp">> => 1678094561525
+                },
+                #{
+                    <<"datatype">> => 2,
+                    <<"int_value">> => 42,
+                    <<"name">> => <<"counter_group1/counter1_10sec">>,
+                    <<"timestamp">> => 1678094561525
+                },
+                #{
+                    <<"datatype">> => 5,
+                    <<"int_value">> => 1,
+                    <<"name">> => <<"counter_group1/counter1_run">>,
+                    <<"timestamp">> => 1678094561525
+                },
+                #{
+                    <<"datatype">> => 5,
+                    <<"int_value">> => 0,
+                    <<"name">> => <<"counter_group1/counter1_reset">>,
+                    <<"timestamp">> => 1678094561525
+                }
+            ],
+        <<"seq">> => 88,
+        <<"timestamp">> => 1678094561521
+    }.
+
+wait_for_sparkplug_schema_registered() ->
+    wait_for_sparkplug_schema_registered(100).
+
+wait_for_sparkplug_schema_registered(0) ->
+    ct:fail("Timed out waiting for sparkplug schema to be registered");
+wait_for_sparkplug_schema_registered(AttemptsLeft) ->
+    case ets:info(?SERDE_TAB, size) of
+        0 ->
+            timer:sleep(100),
+            wait_for_sparkplug_schema_registered(AttemptsLeft - 1);
+        _ ->
+            ok
+    end.
```

Comment:
Nit: maybe we could add a trace point just after `mria:dirty_write(?SERDE_TAB, Serde)` in `do_build_serde` that captures the name and then wait for that?

Reply:
I made this waiting simpler by utilizing snabbkaffe and more precise. See: https://github.com/emqx/emqx/pull/11169/commits/6c2185bed7f5fc0523f4700335320f90b6623c16#diff-692cd92da83a65aa2e2f3e104c475999960f4cadfe836796317c1b93014c7348R792

I don't thing this is worse than inserting a trace point and waiting for that trace point to happen.